src/Persistence/MassTransit.Azure.Cosmos/AzureCosmos/NewtonsoftJsonCosmosClientFactory.cs

```diff
@@ -19,14 +19,16 @@ public class NewtonsoftJsonCosmosClientFactory :
         IDisposable
     {
         readonly CosmosAuthSettings _authSettings;
+        readonly Action<CosmosClientOptions> _cosmosClientOptions;
         readonly ConcurrentDictionary<Type, Lazy<CosmosClient>> _clients;
 
-        public NewtonsoftJsonCosmosClientFactory(CosmosAuthSettings authSettings)
+        public NewtonsoftJsonCosmosClientFactory(CosmosAuthSettings authSettings, Action<CosmosClientOptions> cosmosClientOptions)
         {
             if (authSettings == null)
                 throw new ArgumentNullException(nameof(authSettings));
 
             _authSettings = authSettings;
+            _cosmosClientOptions = cosmosClientOptions;
```

Comment:
I'm not in favor of this approach anymore (I know it is similar to what we have now). But can we use `IOptions<CosmosClientOptions>` so basically if you want it, you can configure through:
```
services.Configure<CosmosClientOptions>()
```
And MT injects it like that, we already have IServiceProvider to create this class, so should be easy to get it, what do you think?

Reply:
I agree with you. I used this approach to be consistent with all the other files. Do you think we should change this way of doing things now? Or can/should we wait for a future (major, as it is probably a breaking change) version of MT to change the approach for all options?

Reply:
For all, is breaking. For new, like this one, I agree use the new/proper pattern same as all the transport options.

Reply:
yeah, I was implying for "this exact change", not for old/everything

Reply:
Ok understood. I will do the change and come back to you when it is ready to be reviewed.

Reply:
The change has been made. I do not really know how you want to manage for `net462` as `IOptions<TOptions>.Value` is not available for this runtime.

![image](https://user-images.githubusercontent.com/130716409/232227059-3f16aac4-f28d-4566-8d78-939dfaf6d6e2.png)